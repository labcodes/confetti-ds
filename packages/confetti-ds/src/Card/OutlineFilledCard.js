import React from "react";
import PropTypes from "prop-types";

import { CardContext } from "./contexts";

export default class OutlineFilledCard extends React.Component {
  static propTypes = {
    /** Components that will be rendered in the Outline Filled Card (CardImage, CardHeader, CardAction, etc.) */
    children: PropTypes.node.isRequired,
    /** Sets Outline Filled Card's colors based on the chosen palette.  */
    color: PropTypes.oneOf(["mineral", "teal", "purple"]).isRequired,
    /** Sets horizontal layout. It changes the position of CardImage. */
    isHorizontal: PropTypes.bool,
    /** Reduces paddings and margins on Card layout. */
    isCompact: PropTypes.bool,
  };

  static defaultProps = {
    isHorizontal: false,
    isCompact: false,
  };

  render() {
    const { children, color, isCompact, isHorizontal } = this.props;

    return (
      <CardContext.Provider value={{ color, cardType: "outlineFilled" }}>
        <article
          className={`lab-card lab-card--outline lab-card--filled lab-card--${color}
          ${isCompact ? " lab-card--compact" : ""}
          ${isHorizontal ? " lab-card--horizontal" : ""}
        `}
        >
          {children}
        </article>
      </CardContext.Provider>
    );
  }
}
