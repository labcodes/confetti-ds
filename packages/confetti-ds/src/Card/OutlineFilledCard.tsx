import React from "react";

import { CardContext } from "./contexts";
import CardHeader from "./CardHeader";

interface OutlineFilledCardProps {
  /** Components that will be rendered in the Card (CardImage, CardHeader, CardAction, etc.) */
  children: React.ReactNode;
  /** Sets Card's colors (mostly background colors) based on the chosen palette.  */
  color: "mineral" | "teal" | "purple";
  /** Sets horizontal layout. It changes the position of CardImage. */
  isHorizontal?: boolean;
  /** Reduces paddings and margins on Card layout. */
  isCompact?: boolean;
  /** Add a class name to make custom changes */
  className?: string;
}

export default function OutlineFilledCard({
  children,
  color,
  isHorizontal = false,
  isCompact = false,
  className = undefined,
}: OutlineFilledCardProps) {
  return (
    <CardContext.Provider value={{ color, cardType: "outlineFilled" }}>
      <article
        className={`lab-card lab-card--outline lab-card--filled lab-card--${color}
      ${isCompact ? " lab-card--compact" : ""}
      ${isHorizontal ? " lab-card--horizontal" : ""}
      ${className ? ` ${className}` : ""}
    `}
      >
        {children}
      </article>
    </CardContext.Provider>
  );
}
