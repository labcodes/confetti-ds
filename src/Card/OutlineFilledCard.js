import React from "react";
import PropTypes from "prop-types";

import { CardContext } from "./contexts";
import { outlineFilledCardChoices } from "./propTypes";

/**
 * @param children components that will be rendered in the Outline Filled Card (CardImage, CardHeader, CardAction, etc.)
 * @param color sets the Outline Filled Card's colors based on the chosen palette.
 * @param isCompact reduces paddings and margins on Card layout.
 * @param isHorizontal sets horizontal layout. It changes the position of CardImage.
 * @param className adds a class name to make custom changes
 * @returns {JSX.Element}
 * @constructor
 */
export default function OutlineFilledCard({
  children,
  color,
  isCompact,
  isHorizontal,
  className,
}) {
  return (
    <CardContext.Provider value={{ color, cardType: "outlineFilled" }}>
      <article
        className={`lab-card lab-card--outline lab-card--filled lab-card--${color}
      ${isCompact ? " lab-card--compact" : ""}
      ${isHorizontal ? " lab-card--horizontal" : ""}
      ${className ? ` ${className}` : ""}
    `}
      >
        {children}
      </article>
    </CardContext.Provider>
  );
}

OutlineFilledCard.propTypes = {
  /** Components that will be rendered in the Outline Filled Card (CardImage, CardHeader, CardAction, etc.) */
  children: PropTypes.node.isRequired,
  /** Sets Outline Filled Card's colors based on the chosen palette.  */
  color: PropTypes.oneOf(outlineFilledCardChoices.color).isRequired,
  /** Sets horizontal layout. It changes the position of CardImage. */
  isHorizontal: PropTypes.bool,
  /** Reduces paddings and margins on Card layout. */
  isCompact: PropTypes.bool,
  /** Add a class name to make custom changes */
  className: PropTypes.string,
};

OutlineFilledCard.defaultProps = {
  isHorizontal: false,
  isCompact: false,
  className: undefined,
};
