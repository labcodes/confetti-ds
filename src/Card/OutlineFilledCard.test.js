import React from "react";
import renderer from "react-test-renderer";

import OutlineFilledCard from "./OutlineFilledCard";

describe("OutlineFilledCard", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<OutlineFilledCard color="teal">Content</OutlineFilledCard>)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with isCompact", async () => {
    const renderedComponent = renderer
      .create(
        <OutlineFilledCard color="teal" skin="pale" isCompact>
          Content
        </OutlineFilledCard>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with isHorizontal", async () => {
    const renderedComponent = renderer
      .create(
        <OutlineFilledCard color="teal" skin="pale" isHorizontal>
          Content
        </OutlineFilledCard>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
