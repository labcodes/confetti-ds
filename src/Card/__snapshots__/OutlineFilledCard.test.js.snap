// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineFilledCard renders with base props 1`] = `
<article
  className="lab-card lab-card--outline lab-card--filled lab-card--teal
          
          
          
        "
>
  Content
</article>
`;

exports[`OutlineFilledCard renders with extra className 1`] = `
<article
  className="lab-card lab-card--outline lab-card--filled lab-card--teal
          
          
           test-classname
        "
>
  Content
</article>
`;

exports[`OutlineFilledCard renders with isCompact 1`] = `
<article
  className="lab-card lab-card--outline lab-card--filled lab-card--teal
           lab-card--compact
          
          
        "
>
  Content
</article>
`;

exports[`OutlineFilledCard renders with isHorizontal 1`] = `
<article
  className="lab-card lab-card--outline lab-card--filled lab-card--teal
          
           lab-card--horizontal
          
        "
>
  Content
</article>
`;
